### PR TITLE
Fix for issue #1844 - Email sent when removing package owner

### PR DESCRIPTION
**Pull request by AlexBHarley**
This PR fixes issue #1844 by sending an email to the package owner when being removed from a package.

I have taken some liberties as I wasn't sure how to phrase the message - feedback on the code and the message would be much appreciated.

### CODE REVIEW

File: src/NuGetGallery/Controllers/JsonApiController.cs [hunk 1]
```diff
@@ -107,8 +107,11 @@ public object RemovePackageOwner(string id, string username)
             {
                 return new { success = false, message = "Owner not found" };
             }
-
             _packageService.RemovePackageOwner(package, user);
+
+            var currentUser = _userService.FindByUsername(HttpContext.User.Identity.Name);
+            _messageService.SendPackageOwnerRemovedNotice(currentUser, user, package);
```
Comment (maartenba) [inline src/NuGetGallery/Controllers/JsonApiController.cs]: You may want to check currentUser for null

File: src/NuGetGallery/Services/MessageService.cs [hunk 2]
```diff
@@ -314,6 +314,36 @@ public void SendPackageOwnerRequest(User fromUser, User toUser, PackageRegistrat
...
+        public void SendPackageOwnerRemovedNotice(User fromUser, User toUser, PackageRegistration package)
+        {
+            if (!toUser.EmailAllowed)
+            {
+                return;
+            }
+            
+            const string subject = "[{0}] The user '{1}' has removed you as an owner of the package '{2}'.";
+
+            var time = DateTime.UtcNow.ToString("MMMM d yyyy");
```
Comment (maartenba) [inline src/NuGetGallery/Services/MessageService.cs]: No need for adding the timestamp - the user's inbox will show when the message was sent anyway.

**Comment by maartenba** [comment]
Looks good! Added a few remarks.

Also can you try squashing your commits into one?
(tip: `git rebase -i HEAD~3` and a `git push --force` to the current branch will solve this :))

**Comment by AlexBHarley** [comment]
Fixed those few things - not sure if my check for null is correct. And squashed into a single commit.

**Comment by maartenba** [comment]
Awesome! Good to go.

**Pull request merged**